### ISSUE TITLE
Bug: Can not export to CSV or TSV at Wekan for Sandstorm

### ISSUE THREAD

**Issue by xet7** [wekan/wekan#3094]
Hi,
using latest from git repo, at Sandstorm:
- Export menu does not work correctly. I tried modifying it, but then:
- Exporting to CSV and TSV did not work. There are some errors about permission denied on downloading those.

@brymut Can you look at this?

You could also test does import work.

Wekan for Sandstorm build info is at:
https://github.com/wekan/wekan-maintainer/wiki/Developing-Wekan-for-Sandstorm

**Comment by xet7** [comment]
I did fix menu so that it shows those options Export JSON/CSV/TSC, but only Export JSON works.

Just guessing, It could be related to what API endpoints are exposed at Sandstorm, there is different export URL for CSV and TSV. But I don't know.

Anyway, I did push newest Wekan to Sandstorm experimental. I would presume most other new features work.

**Comment by brymut** [comment]
Alright, thanks @xet7, I’ll check it out

**Comment by MJMcC9** [comment]
Hi guys,
not sure if I'm adding this to the right thread. 
But first of all thanks for the new features export to csv and tsv! 
My colleagues and I are looking forward to using the feature.

I performed a test on our Wekan as now I'm able to see the new export functions.
Json works, but unfortunatelly csv and tsv seem a bit strange. 
I converted them as txts so you can have a look at them. 

If I somehow can assist in providing more tests or data please let me know.

[wekan-export-board-9dJbn7pZGQd2FQC28_csv.txt](https://github.com/wekan/wekan/files/4649701/wekan-export-board-9dJbn7pZGQd2FQC28_csv.txt)
[wekan-export-board-9dJbn7pZGQd2FQC28_tsv.txt](https://github.com/wekan/wekan/files/4649703/wekan-export-board-9dJbn7pZGQd2FQC28_tsv.txt)

**Comment by brymut** [comment]
Thanks @MJMcC9, appreciate the feedback and test txt. In regard to the result seeming a little strange, could you please elaborate on what you think doesn’t seem right.

**Comment by xet7** [comment]
@MJMcC9 

Those exported csv and tsv files do not contain any real exported content. They look like trying to access csv and tsv export API URLs without logging in, or not having access to those APIs at all, so it only gets normal Wekan HTML for not-logged-in users. I don't know is this Sandstorm preventing access to those API endpoints, or does this require exposing those for Sandstorm.

For example, there is some API path here:
https://github.com/wekan/wekan/blob/master/sandstorm-pkgdef.capnp#L229

And here is how to access export JSON:
https://github.com/wekan/wekan/wiki/Wekan-Sandstorm-cards-to-CSV-using-Python

Problem is, that other parts of API do not currently work #2167

Also, Trello attachments are not currently imported in Sandstorm #2893 

It would be really nice if Wekan and Sandstorm contributors would help with these.

@kentonv @zenhack @ocdtrekkie 

What do you think?

**Comment by MJMcC9** [comment]
Hi @xet7, @brymut 

Thanks for the quick response. My colleague told me that with "snap" a new release of wekan should automatically be installed. I suppose this has happend as we now have the new "export functions". 

@brymut 
I would have expected something like this:
board name, list name, card name, ...
MyNewBoard, MyNewListToDo, MyNewCard1
MyNewBoard, MyNewListDone, MyNewCard2

Attached a screenshot of the board I used for testing:
![ExampleBoard](https://user-images.githubusercontent.com/64065132/82323064-caf08480-99d7-11ea-89df-2cd0618d99a0.png)

Going by xet7's explanation it seems that with only "snap" the endpoints don't extract the boards data correctly. I will check if I can get the remarks from xet7 to work and update you.

The json file seems to be able to reach the endpoint and thus has the expected content:
[wekan-export-board-9dJbn7pZGQd2FQC28_json.txt](https://github.com/wekan/wekan/files/4650170/wekan-export-board-9dJbn7pZGQd2FQC28_json.txt)

**Comment by xet7** [comment]
@MJMcC9 

I think exporting to CSV and TSV does work in Snap, but not in Sandstorm.

**Comment by zenhack** [comment]
Indeed, it doesn't make a lot of sense to use wekan APIs for creating
users in the Sandstorm version, as it doesn't manage that functionality
in the first place.

Unfortunately there's not currently any official programmatic API for
sandstorm meant to be used from outside the system. There was some
discussion of adding something like that here:

    https://github.com/sandstorm-io/sandstorm/issues/2586

...and I'd still like to see it, but it hasn't happened yet.

You could also theoretically use the meteor methods via the DDP socket
directly, but that's not a stable interface and it's not really
documented.

Also worth noting that Sandstorm doesn't really have anything it would
make sense to map a generic "create user" call to, as Sandstorm shells
out to other services for account management for the most part.

-Ian

Quoting Lauri Ojansivu (2020-05-19 07:44:01)

**Comment by MJMcC9** [comment]
> @MJMcC9
> 
> I think exporting to CSV and TSV does work in Snap, but not in Sandstorm.

Hi @xet7  one of my colleagues was able to verify that we definetly run on snap. Hope it doesn't confuse to much if I post our checks here. Let me know if I should create a new issue for this. 
Feel free to move it.

![snapVersion](https://user-images.githubusercontent.com/64065132/82655502-90c6f300-9c22-11ea-8c08-b1daf31e9a99.jpg)

![WekanSnapStatus](https://user-images.githubusercontent.com/64065132/82655321-4ba2c100-9c22-11ea-8c6b-608672001119.png)

Any ideas on how we can further troubleshoot the issue with the export?

**Comment by MJMcC9** [comment]
Hi,
I was able to follow some of the new updates and tried the exports again. 
Noticed that the menu entry for exporting is displayed a bit slower. 
With Chrome I inspected the page and noticed some issues. Can't tell if the information is usefull. Hope it is though :) 

![WekanInspect](https://user-images.githubusercontent.com/64065132/83520336-d23a8680-a4dd-11ea-9860-fecda3b77de6.png)
![WekanInspectAuthToken](https://user-images.githubusercontent.com/64065132/83520361-de264880-a4dd-11ea-9ae1-22389555560c.png)

The authenication token is the same for all. :)
